src/Plutarch/Extra/List.hs

```diff
@@ -1,380 +1,151 @@
+{- | Returns 'PTrue' if the list-like structure contains at least one element.
 
-   @since 1.1.0
+ @since 3.4.0
 -}
 pnotNull ::
-    forall (a :: S -> Type) (s :: S) list.
-    PIsListLike list a =>
-    Term
-        s
-        (list a :--> PBool)
-pnotNull =
-    phoistAcyclic $
-        plam $
-            pelimList (\_ _ -> pcon PTrue) (pcon PFalse)
-
-{- | / O(n) /. Merge two lists which are assumed to be ordered, given a custom comparator.
-    The comparator should return true if first value is less than the second one.
-
-   @since 1.1.0
--}
-pmergeBy ::
-    forall (a :: S -> Type) (s :: S) list.
-    (PIsListLike list a) =>
-    Term
-        s
-        ( (a :--> a :--> PBool)
-            :--> list a
-            :--> list a
-            :--> list a
-        )
-pmergeBy = phoistAcyclic $ pfix #$ plam go
-  where
-    go self comp a b =
-        pif (pnull # a) b $
-            pif (pnull # b) a $
-                unTermCont $ do
-                    ah <- pletC $ phead # a
-                    at <- pletC $ ptail # a
-                    bh <- pletC $ phead # b
-                    bt <- pletC $ ptail # b
-
-                    pure $
-                        pif
-                            (comp # ah # bh)
-                            (pcons # ah #$ self # comp # at # b)
-                            (pcons # bh #$ self # comp # a # bt)
-
-{- | / O(nlogn) /. Merge sort, bottom-up version, given a custom comparator.
-
-   Assuming the comparator returns true if first value is less than the second one,
-    the list elements will be arranged in ascending order, keeping duplicates in the order
-    they appeared in the input.
-
-   @since 1.1.0
--}
-pmsortBy ::
-    forall s a l.
-    (PIsListLike l a, PIsListLike l (l a)) =>
-    Term s ((a :--> a :--> PBool) :--> l a :--> l a)
-pmsortBy = phoistAcyclic $
-    plam $ \comp xs ->
-        mergeAll # comp # (pmap # psingleton # xs)
-  where
-    mergeAll :: Term _ ((a :--> a :--> PBool) :--> l (l a) :--> l a)
-    mergeAll = phoistAcyclic $
-        pfix #$ plam $ \self comp xs ->
-            pif (pnull # xs) pnil $
-                let y = phead # xs
-                    ys = ptail # xs
-                 in pif (pnull # ys) y $
-                        self # comp #$ mergePairs # comp # xs
-    mergePairs :: Term _ ((a :--> a :--> PBool) :--> l (l a) :--> l (l a))
-    mergePairs = phoistAcyclic $
-        pfix #$ plam $ \self comp xs ->
-            pif (pnull # xs) pnil $
-                let y = phead # xs
-                 in plet (ptail # xs) $ \ys ->
-                        pif (pnull # ys) xs $
-                            let z = phead # ys
-                                zs = ptail # ys
-                             in pcons # (pmergeBy # comp # y # z) # (self # comp # zs)
+    forall (a :: S -> Type) (ell :: (S -> Type) -> S -> Type) (s :: S).
+    (PIsListLike ell a) =>
+    Term s (ell a :--> PBool)
+pnotNull = phoistAcyclic $ plam $ \xs -> pnot #$ pnull # xs
```

Comment:
Do we really need this function?

Reply:
I left it in for compatibility. I don't see the harm in having it, but I agree, it's a bit superfluous.